include/dd/Package.hpp

```diff
@@ -1626,13 +1641,15 @@ namespace dd {
 
             ResultEdge e{};
             if constexpr (std::is_same_v<RightOperandNode, mCachedEdge>) {
+                std::cout << "Cached entered \n";
                 // This branch is only taken for matrices
                 if (x.p->v == var && x.p->v == y.p->v) {
+                    std::cout << "Cached if entered \n";
                     if (x.p->isIdentity()) {
                         if constexpr (n == NEDGE) {
                             // additionally check if y is the identity in case of matrix multiplication
                             if (y.p->isIdentity()) {
-                                e = makeIdent(start, var);
+                                // e = makeIdent(start, var);
```

Comment:
I think you need to be careful here when commenting this method out.
if `x` is the identity and `y` is the identity, then `e` is not properly set since you commented out the identity code. In fact, it isn't set at all.